vis_gen.py

```diff
@@ -0,0 +1,565 @@
+
+    @font_style.setter
+    def font_style(self, font_style):
+        """Add style parameter in in freetype font name. No checking if the
+        style exists."""
+        # Do nothing if font_style is None or is an empty string
+        if not font_style:
+            self.__font_style = ""
+            return
+        # Get freetype font and then create Cairo font
+        full_font_name = self.font_name.split(':')[0] + ':style=' + font_style
+        ft_face = self.__ft.find_face(full_font_name)  # temporary
+        self.__font_style = font_style
+        self.__font_face = qah.FontFace.create_for_ft_face(ft_face)
+
+    @antialias.setter
+    def antialias(self, antialias):
+        if antialias not in self.__antialias_map.keys():
+            raise ValueError('Expect antialias to be one of "Default", "Best", '
+                             '"Fast", "Good" or "None".')
+        self.__antialias = self.__antialias_map[antialias]
+
+    @out_dir.setter
+    def out_dir(self, out_dir):
+        self.__out_dir = out_dir
+
+    @grayscale.setter
+    def grayscale(self, grayscale):
+        self.__grayscale = grayscale
+        # Configure format and color according to grayscale option
+        if grayscale:
+            self.__cairo_format = CAIRO.FORMAT_A8 # 8 bits grayscale format
+            self.__canvas_color = Colour.grey(0, 0) # grey(i,a) => rgba(i,i,i,a)
+            self.__text_color = Colour.grey(0, 1) # Only alpha value matters
+        else:
+            self.__cairo_format = CAIRO.FORMAT_RGB24
+            self.__canvas_color = Colour.grey(255, 1)
+            self.__text_color = Colour.grey(0, 1)
+            # If /usr/share/X11/rgb.txt exists, color can be specified by
+            # Colour.x11['colour_name']
+
+    def generate_dataset_from_file(self, file_path, font_styles, antialiases):
+        """Read code points from file and visualize. Character set file must
+        follow this format:
+            1. Each line represents a single code point
+            2. Each code point is in format 'U+2a665'
+
+        Args:
+            file_path: Str, path to file for the character set.
+            font_styles: list of Str, styles of the font face to visualize
+            antialiases: list of Str, antialiasing styles to visualize
+        """
+        # Check if out_dir exists and create if not
+        out_dir_abs = os.path.join(os.getcwd(), self.out_dir)
+        if self.__check_out_dir:
+            self.__check_create_out_dir_abs(out_dir_abs)
+        self.__check_out_dir = False
+
+        # Read file and translate format
+        code_points = []
+        with open(file_path) as f:
+            for line in f:
+                code_point = chr(int('0x' + line.split('\n')[0][2:], 16))
+                code_points.append(code_point)
+
+        self.generate_dataset_from_list(code_points, font_styles, antialiases)
+
+
+    def generate_dataset_from_list(self, code_points, font_styles, antialiases):
```

Comment:
Please write test. Mock out the visualize_list to verify generate_dataset_from_list will call visualize_list with the right number of time with the right parameters.